cratedb_toolkit/testing/testcontainers/cratedb.py

```diff
@@ -51,44 +57,87 @@ class CrateDBContainer(KeepaliveContainer, DbContainer):
     CRATEDB_PASSWORD = os.environ.get("CRATEDB_PASSWORD", "")
     CRATEDB_DB = os.environ.get("CRATEDB_DB", "doc")
     KEEPALIVE = asbool(os.environ.get("CRATEDB_KEEPALIVE", os.environ.get("TC_KEEPALIVE", False)))
+    CMD_OPTS = {
+        "discovery.type": "single-node",
+        "node.attr.storage": "hot",
+        "path.repo": "/tmp/snapshots",
+    }
 
-    # TODO: Dual-port use with 4200+5432.
     def __init__(
         self,
         image: str = "crate/crate:nightly",
-        port: int = 4200,
+        ports: Optional[dict] = None,
         user: Optional[str] = None,
         password: Optional[str] = None,
         dbname: Optional[str] = None,
         dialect: str = "crate",
+        cmd_opts: Optional[dict] = None,
         **kwargs,
     ) -> None:
+        """
+        :param image: docker hub image path with optional tag
+        :param ports: optional dict that maps a port inside the container to a port on the host machine;
+                      `None` as a map value generates a random port;
+                      Dicts are ordered. By convention the first key-val pair is designated to the HTTP interface.
+                      Example: {4200: None, 5432: 15431} - port 4200 inside the container will be mapped
+                      to a random port on the host, internal port 5432 for PSQL interface will be mapped
+                      to the 15432 port on the host.
+        :param user:  optional username to access the DB; if None, try respective environment variable
+        :param password: optional password to access the DB; if None, try respective environment variable
+        :param dbname: optional database name to access the DB; if None, try respective environment variable
+        :param dialect: a string with the dialect to generate a DB URI
```

Comment:
why do we want to expose `dialect` as parameter? is there a use case to change it?

Reply:
I am not 100% sure, but I think when passed `"http"` instead of `None`, the corresponding utility function will return the HTTP-based connection URL, and not the SQLAlchemy-based connection URL, in the same manner as `get_http_url()` does it.

The `dialect` argument has already been part of the implementation before the refactoring, and I think it is part of the official/designated constructor signature. See [^1].

[^1]: https://github.com/crate-workbench/cratedb-toolkit/blob/v0.0.2/cratedb_toolkit/testing/testcontainers/cratedb.py#L61

Reply:
`dialect` is normally used to use a DB specific dialect instead of a more generic one, e.g. psql vs jdbc, it's what we use to set `crate` instead of `psql` for the cases that we want to use the `crate` jdbc driver instead of the psql one. `http` vs binary `psql` protocol is not a dialect, but a protocol.

Reply:
Not saying that it's part of the PR, just mentioning that it seems weird to me to expose a "dialect".

Reply:
Actually, I've double checked the dialect thing and you are right @matriv - no need to have it in the class constructor. It's enough to have it in the `get_connection_url` as an argument with the default value `crate`. Gonna fix that